### PR TITLE
VACMS-17854: Add feature toggle to display old/new telephone fields conditionally

### PR DESCRIPTION
**Pull request by dsasser**
## Description
Moves the new telephone migration fields behind a feature toggle.

Relates to #17854

## Testing done
Manual

### Run the migration (daily, once; check the [telephone audit View](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/content/audit/phone-numbers) and/or the [Codit Batch Operations logs](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/config/development/batch_operations/log) to see if the migration has been run by someone else before running)
Get a [terminal](https://tugboat.vfs.va.gov/tush/6706e93425a4ad7d7680df68) in the tugboat
Run the following commands:

```
time drush codit-batch-operations:run MigratePhoneFieldToParagraph &&
time drush codit-batch-operations:run MigrateVamcBillingInsurancePhoneFieldToParagraph &&
time drush codit-batch-operations:run MigrateVamcFacilityMentalHealthPhoneFieldToParagraph 
```
### Validate Old Fields Appear on Node New with Toggle Off:
- [x] Visit the [Flipper Toggle admin page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/config/system/feature_toggle) and ensure the `FEATURE_TELEPHONE_MIGRATION_V1` toggle is not checked.
- [x] View a [VAMC System Billing and Insurance node](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/45681) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] View a [Staff Profile](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/73150) in new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] View a [VAMC Facility](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/71154) in a new tab (you will return to this tab later)
- [x] Confirm the Mental health old phone number is visible
- [x] Confirm the new phone number is not visible

### Validate Old Fields Appear on Node Edit with Toggle Off:
- [x] Visit the [Flipper Toggle admin page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/config/system/feature_toggle) and ensure the `FEATURE_TELEPHONE_MIGRATION_V1` toggle is not checked.
- [x] Edit a [VAMC System Billing and Insurance node](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/45681/edit) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] Edit a [Staff Profile](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/73150/edit) in new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] Edit a [VAMC Facility](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/71154/edit) in a new tab (you will return to this tab later)
- [x] Confirm the old Mental health phone number is visible
- [x] Confirm the new phone number is not visible

### Validate Old Fields appear on Node Add Forms with Toggle Off:
- [x] Visit the [create VAMC System Billing and Insurance node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/vamc_system_billing_insurance) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] Visit the [create Staff Profile node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/person_profile) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] Visit the [create VAMC Facility node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/health_care_local_facility) in a new tab (you will return to this tab later)
- [x] Confirm the old Mental health phone number is visible
- [x] Confirm the new phone number is not visible

### Validate New Fields Appear on Node View With Toggle On
- [x] Visit the [Flipper Toggle admin page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/config/system/feature_toggle) and check the `FEATURE_TELEPHONE_MIGRATION_V1` checkbox and save the form.
- [x] Clear the cache
- [x] View the [VAMC System Billing and Insurance node](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/45681)
- [x] Confirm the new phone number is visible
- [x] Confirm the old phone number not visible
- [x] View the [Staff Profile](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/73150)
- [x] Confirm the new phone number is visible
- [x] Confirm the old phone number not visible
- [x] View a [VAMC Facility](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/71154) 
- [x] Confirm the new Mental health phone number is visible
- [x] Confirm the old phone number not visible

### Validate New Fields Appear on Node Edit With Toggle On
- [x] Visit the [Flipper Toggle admin page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/admin/config/system/feature_toggle) and check the `FEATURE_TELEPHONE_MIGRATION_V1` checkbox and save the form.
- [x] Return to the [VAMC System Billing and Insurance node](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/45681/edit)
- [x] Confirm the new phone number is visible
- [x] Confirm the old phone number not visible
- [x] Return to the [Staff Profile](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/73150/edit) in new tab
- [x] Confirm the new phone number is visible
- [x] Confirm the old phone number not visible
- [x] Return to the [VAMC Facility](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/71154/edit)
- [x] Confirm the new Mental health phone number is visible
- [x] Confirm the old Mental health phone number not visible

### Validate New Fields appear on Node Add Forms with Toggle On:
- [x] Visit the [create VAMC System Billing and Insurance node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/vamc_system_billing_insurance) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [x] Visit the [create Staff Profile node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/person_profile) in a new tab (you will return to this tab later)
- [x] Confirm the old phone number is visible
- [x] Confirm the new phone number is not visible
- [ ] Visit the [create VAMC Facility node page](https://pr19472-t0jmxpsbdevdalhcuolnvbok7utptd4d.ci.cms.va.gov/node/add/health_care_local_facility) in a new tab (you will return to this tab later)
- [ ] Confirm the old Mental health phone number is visible
- [ ] Confirm the new phone number is not visible

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [x] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

**Comment by omahane** [comment]
As soon as I went to look at the node on the node:view (as opposed to node:edit), I noticed that the new field is showing even with the toggle off. 

I believe that we would expect to find the old field showing on both the view and edit screens.

Screenshot 2024-10-10 at 4 06 03 PM

**Comment by omahane** [comment]
# Node:edit (Toggle on), as content admin
I refactored the form display a bit to account for the fact that we have 2 phone fields on the VAMC facility, as I saw that we ended up hiding the facility phone number on the form for content admins (and above) unintentionally:
Screenshot 2024-10-11 at 10 33 02 AM (2)

**Comment by omahane** [comment]
We should check 
- node:view for all three content types
- admins
- non-admins

**Comment by davidmpickett** [comment]
@omahane I'm seeing both Old & New fields on Node Edit for VAMC System Billing and Insurance with Toggle On
Screenshot 2024-10-14 at 4 19 04 PM

**Comment by davidmpickett** [comment]
Verified that toggle is behaving as expected in Tugboat

**Review by davidmpickett — APPROVED** [review]
Toggle works as expected in all scenarios.

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php [hunk 1]
```diff
@@ -40,6 +46,34 @@ class PostFacilityStatus extends PostFacilityBase implements PostServiceInterfac
    */
   protected $additionalInfoToPush;
 
+  /**
```
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php:49]: Feature toggle function

File: docroot/modules/custom/va_gov_profile/va_gov_profile.services.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 services:
   va_gov_profile.entity_event_subscriber:
     class: Drupal\va_gov_profile\EventSubscriber\EntityEventSubscriber
-    arguments: ['@va_gov_user.user_perms', '@entity_type.manager']
+    arguments: ['@va_gov_user.user_perms', '@entity_type.manager', '@feature_toggle.feature_status']
```
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_profile/va_gov_profile.services.yml:4]: adds feature toggle to va profile services

File: docroot/modules/custom/va_gov_vamc/src/EventSubscriber/VAMCEntityEventSubscriber.php [hunk 3]
```diff
@@ -362,6 +402,31 @@ public function alterTopTaskNodeForm(FormIdAlterEvent $event): void {
   public function alterVamcSystemBillingAndInsuranceForm(FormIdAlterEvent $event) {
     $this->alterTopTaskNodeForm($event);
     $this->removePhoneLabel($event);
+    $this->showTelephone($event);
```
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_vamc/src/EventSubscriber/VAMCEntityEventSubscriber.php:405]: logic to show correct phone field from feature toggle

**Review by jv-agile6 — APPROVED** [review]
Reviewed code for logic and form, no defects identified. Approved.

**Pull request merged**